---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

```{r}
library(gggenes)
library(tidyverse)
data<-read_tsv("neb_vss1a.primer.bed.tsv", col_names=FALSE)
data
colnames(data)=c("chromosome","start","end","name","X5","strand")
data = data %>% separate(name,into=c("library","number","pos"),sep="_") 



readF = function(filename){
cov <- read_tsv(paste0("coverage/",filename),col_names=F)
colnames(cov)=c("gb","pos","depth")
cov$file = filename
cov = cov %>% filter(pos>20900,pos<23500)
cov$depth =cov$depth/max(cov$depth)
return(cov)
}


files = list.files("coverage")

dfs = lapply(files,readF)
big = bind_rows(dfs)
library(zoo)

big = big %>% group_by(file) %>% arrange(pos) %>%  mutate(meaned=rollmean(depth,k=21,na.pad=T))
  
  region_start = 22673
region_end = 22687


ggplot(big%>%mutate(run_label=gsub(".coverage","",gsub(".fastq","",file)) %>% filter(
  run_label!="SRR19187122" # just to get an even number for plotting in a grid
                                                                                     
                                                                                     )),aes(x=pos,y=meaned,color=file)) +geom_line() + coord_cartesian(xlim=c(21900,23500))+theme_bw() + theme(legend.position = "none") +facet_wrap(~run_label)+scale_y_continuous(expand=c(0,0),breaks=c(0,0.5,1),limits=c(0,1), label=scales::percent)+annotate("point",x=(region_start+region_end)/2,y=0.1,color="gray" )+labs(x="Genome position",y="Normalised coverage")



b<-ggplot(data %>% filter(start>20900,end<23500), aes(xmin = start, xmax = end,y="1",forward=strand=="+",label=number,fill=number)) +
  geom_gene_arrow()  +geom_gene_label() +
  scale_fill_brewer(palette = "Set3")+
  theme_genes()  + coord_cartesian(xlim=c(20900,23500))

library(patchwork)

a/b


```

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Ctrl+Alt+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Ctrl+Shift+K* to preview the HTML file).

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.
